AsyncDisplayKit/ASCellNode.m

```diff
@@ -32,6 +40,35 @@ - (instancetype)init
   return self;
 }
 
+- (instancetype)initWithViewControllerBlock:(ASDisplayNodeViewControllerBlock)viewControllerBlock
+{
+  return [self initWithViewControllerBlock:viewControllerBlock didLoadBlock:nil];
+}
+
+- (instancetype)initWithViewControllerBlock:(ASDisplayNodeViewControllerBlock)viewControllerBlock didLoadBlock:(ASDisplayNodeDidLoadBlock)didLoadBlock
+{
+  if (!(self = [super init]))
+    return nil;
+  
+  ASDisplayNodeAssertNotNil(viewControllerBlock, @"should initialize with a valid block that returns a UIViewController");
+  
+  _viewController = viewControllerBlock();
+  
+  _viewControllerNode = [[ASDisplayNode alloc] initWithViewBlock:^UIView *{
+        return _viewController.view;
+      } didLoadBlock:didLoadBlock];
+  
+  [self addSubnode:_viewControllerNode];
+  
+  _nodeLoadedBlock = didLoadBlock;
+  return self;
+}
+
+- (ASLayoutSpec *)layoutSpecThatFits:(ASSizeRange)constrainedSize {
```

Comment:
Alright, I went through your comments and fixed everything except for this so far.  For some reason the constrainedSize.max returns the correct size for each cell but when I try to just set the frame in -layout or -layoutDidFinish every cell returns a size of (width = 355, height = 44).

Seemingly unrelated, once you get to those methods _viewControllerNode is nil for some reason.